punishments.js

```diff
@@ -667,18 +683,30 @@ Punishments.roomBan = function (room, user, expireTime, id, ...rest) {
 };
 
 Punishments.roomBlacklist = function (room, user, expireTime, id, ...rest) {
-	if (!id) id = user.getLastId();
+	if (!id && user) id = user.getLastId();
 
 	if (!expireTime) expireTime = Date.now() + BLACKLIST_DURATION;
 	let punishment = ['BLACKLIST', id, expireTime].concat(rest);
-	Punishments.roomPunish(room, user, punishment);
+	if (user) {
```

Comment:
The thing is however that I want it to remain possible to blacklistname a school IP and not blacklist the entire school for a year.